src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  // Rozdzielamy ciąg na linie, usuwamy puste linie
+  // i linie zawierające tylko białe znaki
+  const lines = sourceString.split(';').filter((line) => line.trim());
+
+  // Używamy reduce do przekształcenia tablicy linii w obiekt stylów
+  return lines.reduce((stylesObject, line) => {
+    // Używamy wyrażenia regularnego do wyodrębnienia klucza i wartości z linii
+    const match = line.match(/^\s*([^:]+?)\s*:\s*([\s\S]+?)\s*$/);
```

Comment:
Here i would just simply use split method instead of regex. Its less performant and more complex code to read.